src/xdp-utils.c

```diff
@@ -1094,17 +1096,28 @@ verify_proc_self_fd (XdpAppInfo *app_info,
      but there is not much to do about this. */
   if (g_str_has_suffix (path_buffer, " (deleted)"))
     {
-      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_FILENAME,
-                   "Cannot share deleted file: %s", path_buffer);
-      return NULL;
+      if (documents_mountpoint != NULL &&
+          g_str_has_prefix(path_buffer, documents_mountpoint))
+        {
+          /* Unfortunately our workaround for dcache purging triggers
+             o_path file descriptors on the fuse filesystem being
+             marked as deleted, so we have to allow these here and
+             rewrite them. This is safe, becase we will stat the file
+             and compare to make sure we end up on the right file. */
+          path_buffer[symlink_size - strlen(" (deleted)")] = 0;
```

Comment:
Perhaps it's worth asserting `g_assert (symlink_size > strlen (" (deleted)")`?

Also, perhaps it's clearer to pass an explicit character here: `\0`

Reply:
Can't we make `strlen(" (deleted)")` a constant instead? I don't think the compile will optimize that

Reply:
We just verified that it has " (deleted)" as a suffix. Not sure why it would not fit in there?

@hfiguiere GCC will in fact optiize strlen of constant strings.

Reply:
```suggestion
          path_buffer[symlink_size - strlen (" (deleted)")] = 0;
```